src/System.Threading.Thread/tests/ThreadTests.cs

```diff
@@ -139,6 +144,85 @@ yield return
                 };
         }
 
+        [Theory]
+        [InlineData("STAMain.exe", "GetApartmentState")] 
+        [InlineData("STAMain.exe", "SetApartmentState")]
+        [InlineData("MTAMain.exe", "GetApartmentState")]
+        [InlineData("MTAMain.exe", "SetApartmentState")]
+        [ActiveIssue(20766, TargetFrameworkMonikers.Uap)]
+        public static void ApartmentState_AtributePresent(string AppName, string mode)
+        {
+            var psi = new ProcessStartInfo();
+            if (PlatformDetection.IsFullFramework || PlatformDetection.IsNetNative)
+            {
+                psi.FileName = AppName;
+                psi.Arguments = $"{mode}";
+            }
+            else
+            {
+                psi.FileName = DummyClass.HostRunnerTest;
+                psi.Arguments = $"{AppName} {mode}";
+            }
+            using (Process p = Process.Start(psi))
+            {
+                p.WaitForExit();
+                if (PlatformDetection.IsWindows)
+                {
+                    Assert.Equal(0, p.ExitCode);
+                }
+                else
+                {
+                    Assert.Equal(1, p.ExitCode);
+                }
+            }
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        public static void ApartmentState_NoAttributePresent_DefaultState_Windows()
+        {
+            DummyClass.RemoteInvoke(() =>
+            {
+                Assert.Equal(ApartmentState.MTA, Thread.CurrentThread.GetApartmentState());
+                Assert.Throws<InvalidOperationException>(() => Thread.CurrentThread.SetApartmentState(ApartmentState.STA));
+                Thread.CurrentThread.SetApartmentState(ApartmentState.MTA);
+            }).Dispose();
+        }
+
+        [Fact]
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
```

Comment:
Needs a comment explaining why you skip on .NET Framework. Or (better?) another test that is .NET Framework only that demonstrates the difference.